### ISSUE TITLE
Provider produced Inconsistent response on Config Aggregators

### ISSUE THREAD

**Issue by bo1984** [terraform-providers/terraform-provider-aws#13375]
### Terraform Version

Terraform v0.12.25
+ provider.aws v2.62.0

### Affected Resource(s)

Link to full module - https://www.dropbox.com/s/jtq8oil4z7xbi54/terraform_debug.zip?dl=0

```
resource "aws_config_configuration_aggregator" "organization" {
  name = "OrganizationAggregation" # Required

  organization_aggregation_source {
    all_regions = true
    role_arn    = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/OrganizationConfigRole"
  }
}
```
### Terraform Configuration Files

```
provider.tf

provider aws {
  region = "us-east-1" 
  shared_credentials_file = "/Users/lechange/.aws/credentials"
  profile                 = "default"
}

provider aws {
  alias  = "us-east-2"
  region = "us-east-2"
  shared_credentials_file = "/Users/lechange/.aws/credentials"
  profile                 = "default"
}
```

```
main.tf

module "us-east-1" {
  source = "./config"

  providers = {
    aws = aws
  }
}

module "us-east-2" {
  source = "./config"

  providers = {
    aws = aws.us-east-2
  }
}
```
Affected resource:
```
resource "aws_config_configuration_aggregator" "organization" {
  name = "OrganizationAggregation" # Required

  organization_aggregation_source {
    all_regions = true
    role_arn    = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/OrganizationConfigRole"
  }
}
```

### Debug Output

https://gist.github.com/bo1984/79a4ee95099e53ae12b12addef799043

### Expected Behavior

Creating a multi-region TF module to create AWS Config Recorders and delivery channels in us-east-1 and us-east-2, respectively. 

### Actual Behavior

The stack launches fine, up until I include an account-based or organization-based Config aggregator (aws_config_configuration_aggregator). This appears to be a soft error as described in the TF documentation. I get the following error: 

```
aws_config_configuration_aggregator.organization: Creating...

Error: Provider produced inconsistent result after apply

When applying changes to aws_config_configuration_aggregator.organization,
provider "registry.terraform.io/-/aws" produced an unexpected new value for
was present, but now absent.

This is a bug in the provider, which should be reported in the provider's own
issue tracker.
```

When running an apply, it appears that the TF state, given the aforementioned error, sees that it needs to create the aggregator, even though it already exists. I've attempted to remove it from the tfstate file along with deleting it from the account. I've even created a completely separate TF module just for the aggregator and still get an error for it as depicted above. No matter the approach, I still get an error trying to create the aggregator (which still shows up in the account).

```
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # aws_config_configuration_aggregator.organization will be created
  + resource "aws_config_configuration_aggregator" "organization" {
      + arn  = (known after apply)
      + id   = (known after apply)
      + name = "OrganizationAggregation"

      + organization_aggregation_source {
          + all_regions = true
          + role_arn    = "arn:aws:iam::725369550382:role/OrganizationConfigRole"
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.
```

### Steps to Reproduce

1. `terraform apply`

**Comment by ewbankkit** [comment]
@bo1984 Thanks for raising this issue.
This class of errors is usually caused by eventual-consistency issues in the AWS service's API. The resource is successfully created but when Terraform turns around to read the resource, the resource is not (yet) found.
The solution is often to add a read-retry loop as part of the resource creation.

**Comment by bo1984** [comment]
@ewbankkit Thank you for providing me with an alternative. Are there any public referenceable  examples for implementing this? I've tried some of Terraform's documentation and am unsuccessful in my efforts so far.

**Comment by bo1984** [comment]
@ewbankkit as I typed my last message, I remember I came across this back in February as I looked this up previously. The message is a bit of a red herring, but the config aggregator name should be allow lowercase alphabetic characters. When you put in capitals, you run in this issue as I ran across this link a while back - https://github.com/terraform-providers/terraform-provider-aws/issues/11769 . Thanks again for your help.
